packages/amplify-cli-core/src/utils/validate-path.ts

```diff
@@ -1,22 +1,23 @@
 import * as fs from 'fs-extra';
 import { ExportPathValidationError } from '../errors';
-
+import * as path from 'path';
 /**
  * Validates whether the path is a directory
  * @throws {ExportPathValidationError} if path not valid
  * @param directoryPath to validate
  */
-export function validateExportDirectoryPath(directoryPath: any) {
-  if (typeof directoryPath !== 'string') {
-    throw new ExportPathValidationError(`${directoryPath} is not a valid path specified by --out`);
-  }
+export function validateExportDirectoryPath(directoryPath: any, defaultPath: string): string {
+  const exportPath = directoryPath || defaultPath;
+  const resolvedDir = path.resolve(exportPath);
 
-  if (!fs.existsSync(directoryPath)) {
-    throw new ExportPathValidationError(`${directoryPath} does not exist`);
+  if (!fs.existsSync(resolvedDir)) {
+    fs.ensureDirSync(resolvedDir);
```

Comment:
nit: you can call `fs.ensureDirSync(resolvedDir);` without checking if the folder exists already.

Reply:
check the else